tests/TestCase/Database/Driver/MysqlTest.php

```diff
@@ -47,7 +47,7 @@ public function setup()
      */
     public function testConnectionConfigDefault()
     {
-        $driver = $this->getMock('Cake\Database\Driver\Mysql', ['_connect']);
+        $driver = $this->getMock('Cake\Database\Driver\Mysql', ['_connect', 'connection']);
         $dsn = 'mysql:host=localhost;port=3306;dbname=cake;charset=utf8';
```

Comment:
One common way is saving data from another connection that had not the same encoding. Or creating the database with another encoding and trying to read it as utf8